app/src/androidTest/java/com/vero/tasky/core/presentation/navigation/RootNavigationTest.kt

```diff
@@ -0,0 +1,244 @@
+
+import android.content.Intent.ACTION_VIEW
+import androidx.compose.material.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation.NavHostController
+import androidx.navigation.NavType
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.navArgument
+import androidx.navigation.navDeepLink
+import com.vero.tasky.R
+import com.vero.tasky.agenda.domain.model.AgendaItem
+import com.vero.tasky.agenda.domain.model.AgendaItemType
+import com.vero.tasky.agenda.presentation.agenda.AgendaScreen
+import com.vero.tasky.agenda.presentation.eventdetails.EventDetailsScreen
+import com.vero.tasky.agenda.presentation.editphoto.EditPhotoScreen
+import com.vero.tasky.agenda.presentation.edittext.EditTextScreen
+import com.vero.tasky.agenda.presentation.reminderdetails.ReminderDetailsScreen
+import com.vero.tasky.agenda.presentation.taskdetails.TaskDetailsScreen
+import com.vero.tasky.auth.presentation.login.LoginScreen
+import com.vero.tasky.auth.presentation.login.LoginViewModel
+import com.vero.tasky.auth.presentation.registration.RegistrationScreen
+import com.vero.tasky.core.presentation.navigation.NavigationConstants.Companion.EDIT_PHOTO_URI
+import com.vero.tasky.core.presentation.navigation.NavigationConstants.Companion.EDIT_TEXT_TYPE
+import com.vero.tasky.core.presentation.navigation.NavigationConstants.Companion.EDIT_TEXT_VALUE
+import com.vero.tasky.core.presentation.navigation.NavigationConstants.Companion.EVENT_ITEM_ID
+import com.vero.tasky.core.presentation.navigation.NavigationConstants.Companion.IS_EDITABLE
+import com.vero.tasky.core.presentation.navigation.NavigationConstants.Companion.REMINDER_ITEM_ID
+import com.vero.tasky.core.presentation.navigation.NavigationConstants.Companion.TASK_ITEM_ID
+import com.vero.tasky.core.presentation.navigation.NavigationConstants.Companion.getAgendaItemScreenArgumentTypes
+import com.vero.tasky.core.presentation.navigation.NavigationConstants.Companion.getAgendaItemScreenParameters
+import com.vero.tasky.core.presentation.navigation.NavigationConstants.Companion.getEventDeepLink
+import com.vero.tasky.core.presentation.navigation.NavigationConstants.Companion.getReminderDeepLink
+import com.vero.tasky.core.presentation.navigation.NavigationConstants.Companion.getTaskDeepLink
+import com.vero.tasky.ui.theme.Inter400Size18
+import com.vero.tasky.ui.theme.Inter400Size26
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+
+@Composable
+fun RootNavigationTest(
+    navController: NavHostController,
+    isLoggedIn: Boolean,
+    modifier: Modifier = Modifier,
+    loginViewModel: LoginViewModel? = null
+) {
+
+    NavHost(
+        navController = navController,
+        startDestination = if (isLoggedIn) Screens.Agenda.route else Screens.Login.route,
+        modifier = modifier
+    ) {
```

Comment:
Same here, use the NavHost from your production code instead and don't recreate it. This is very error prone and won't catch issues in your real NavHost